### PR TITLE
Add documentation on configuring Transport Layer Security in Egeria

### LINKED ISSUE #3705: Egeria chassis SSL context error - update docs
The egeria chassis fails to launch in a default configuration when run from the 'target' directory.

```
jonesn:server-chassis-spring/ (master) $ LOGGING_LEVEL_ROOT=Info java -jar target/server-chassis-spring-2.4-SNAPSHOT.jar
 ODPi Egeria
    ____   __  ___ ___    ______   _____                                 ____   _         _     ___
   / __ \ /  |/  //   |  / ____/  / ___/ ___   ____ _   __ ___   ____   / _  \ / / __    / /  / _ /__   ____ _  _
  / / / // /|_/ // /| | / / __    \__ \ / _ \ / __/| | / // _ \ / __/  / /_/ // //   |  / _\ / /_ /  | /  _// || |
 / /_/ // /  / // ___ |/ /_/ /   ___/ //  __// /   | |/ //  __// /    /  __ // // /  \ / /_ /  _// / // /  / / / /
 \____//_/  /_//_/  |_|\____/   /____/ \___//_/    |___/ \___//_/    /_/    /_/ \__/\//___//_/   \__//_/  /_/ /_/

 :: Powered by Spring Boot (v2.3.3.RELEASE) ::

15:39:33.833 [main] INFO  o.o.o.s.s.OMAGServerPlatform - Starting OMAGServerPlatform on planetf1mac with PID 73802 (/Users/jonesn/IdeaProjects/egeria-maven/open-metadata-implementation/server-chassis/server-chassis-spring/target/server-chassis-spring-2.4-SNAPSHOT.jar started by jonesn in /Users/jonesn/IdeaProjects/egeria-maven/open-metadata-implementation/server-chassis/server-chassis-spring)
15:39:33.840 [main] INFO  o.o.o.s.s.OMAGServerPlatform - No active profile set, falling back to default profiles: default
15:39:38.333 [main] INFO  o.s.b.w.e.tomcat.TomcatWebServer - Tomcat initialized with port(s): 9443 (https)
15:39:38.367 [main] INFO  o.a.coyote.http11.Http11NioProtocol - Initializing ProtocolHandler ["https-jsse-nio-9443"]
15:39:38.367 [main] INFO  o.a.catalina.core.StandardService - Starting service [Tomcat]
15:39:38.368 [main] INFO  o.a.catalina.core.StandardEngine - Starting Servlet engine: [Apache Tomcat/9.0.37]
15:39:38.651 [main] INFO  o.a.c.c.C.[Tomcat].[localhost].[/] - Initializing Spring embedded WebApplicationContext
15:39:41.950 [main] INFO  o.a.catalina.core.StandardService - Stopping service [Tomcat]
15:39:41.980 [main] ERROR o.s.boot.SpringApplication - Application run failed
org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'healthIndicatorRegistry' defined in class path resource [org/springframework/boot/actuate/autoconfigure/health/LegacyHealthEndpointCompatibilityConfiguration.class]: Unsatisfied dependency expressed through method 'healthIndicatorRegistry' parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'healthContributorRegistry' defined in class path resource [org/springframework/boot/actuate/autoconfigure/health/HealthEndpointConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.springframework.boot.actuate.health.HealthContributorRegistry]: Factory method 'healthContributorRegistry' threw exception; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'elasticsearchHealthContributor' defined in class path resource [org/springframework/boot/actuate/autoconfigure/elasticsearch/ElasticSearchRestHealthContributorAutoConfiguration.class]: Unsatisfied dependency expressed through method 'elasticsearchHealthContributor' parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'elasticsearchRestClient' defined in class path resource [org/springframework/boot/autoconfigure/elasticsearch/ElasticsearchRestClientConfigurations$RestClientFallbackConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.elasticsearch.client.RestClient]: Factory method 'elasticsearchRestClient' threw exception; nested exception is java.lang.IllegalStateException: could not create the default ssl context
```
The full log is at https://gist.github.com/ffff1f08a18853f9d01297a57dce4075

The root issue appears to be `Caused by: java.lang.IllegalStateException: could not create the default ssl context` - but this is occuring when the java is launched directly ie:
```
LOGGING_LEVEL_ROOT=Info java -jar target/server-chassis-spring-2.4-SNAPSHOT.jar
```
which previously worked.

This may be a classpath issue or a change in default configuration

Specifically it's possible spring boot actuator is causing a problem here -- for example we see references to 'health contributor' and specifically `org/springframework/boot/actuate/autoconfigure/elasticsearch/ElasticSearchRestHealthContributorAutoConfiguration.class`

### PR DESCRIPTION
**Pull request by planetf1**
Fixes #3705

* Script to generate self signed certificate authority & client/server certs compatible with Egeria and current browsers
* README covering the required spring and java configuration options for keystores and truststores
* pictures explaining 1way and 2way SSL
* sample fragment included to launch chassis

I've placed these under open-metadata-resources/open-metadata-deployment/certificates for now but open to suggestions on better places & additional links to add.

There's no question these docs could be improved, but hopefully this is a starting point.

For example additions could cover
 - specific instructions for python, go, 
 - use of the spring config file 
 - specifics on the UIs

Contributions welcome (especially from UI owners)! I'd really like to at least get some docs down ...

It's also worth pointing out this is to meant to document how things work right now. There is no functional change in this issue

### CODE REVIEW

File: open-metadata-resources/open-metadata-deployment/certificates/README.md [hunk 1]
```diff
@@ -1,17 +1,43 @@
 <!-- SPDX-License-Identifier: CC-BY-4.0 -->
 <!-- Copyright Contributors to the Egeria project. -->
```
Comment (mandy-chessell) [inline open-metadata-resources/open-metadata-deployment/certificates/README.md:3]: This README is hidden - it requires knowledge of the project structure to find it.  It should be part of the Admin Guide (around configuring the platform).

**Review by mandy-chessell — CHANGES_REQUESTED** [review]

File: open-metadata-resources/open-metadata-deployment/certificates/README.md (hunk 1 above)
Comment (planetf1) [reply open-metadata-resources/open-metadata-deployment/certificates/README.md:3]: Thanks. 

Will move most content to
 https://github.com/odpi/egeria/blob/master/open-metadata-implementation/admin-services/docs/user/configuring-the-omag-server-platform.md with a link back to the directory originally in this PR just for the certificate creation artifacts

**Comment by planetf1** [comment]
Having looked again, it's not entirely clear where the best place to add this into the documentation is

* The top level entry page is in open-metadata-publication/website/omag-server/README.md - this refers to a tar file - but actually points to the build tree for creating the tar files. However the subsequent task is correct and refers to doing a full build - this is probably the entry point only for some users
*In this README it doesn't exactly say what to launch for the server, though the previous step hints it's probably the jar
* The implication is to start it.. 

Just before here is probably the best place to insert the security docs - as what follows refers to the Admin Services User guide - which are services offered by the platform, rather than the launch configuration of the platform -- yet in the linked to doc, there we have info on server properties, which logically is the same aspect as configuring TLS

As such my approach - initially - is going to be to just add a link into this README to the files in the PR (in the certificates directory). I do think some further refactoring is needed, but we probably need to revisit in a more complete way to talk about build vs assemblies vs docker vs charts in terms of deployment.. as the approach will differ across those.

**Comment by planetf1** [comment]
Content now moved to admin guide

**Pull request merged**